pkgs/by-name/ed/edopro/package.nix

```diff
@@ -0,0 +1,262 @@
+        --bundled-font="${font}" \
+        --covers=\"${covers_url}\" \
+        --fields=\"${fields_url}\" \
+        --no-core \
+        --no-joystick=false \
+        --pics=\"${pics_url}\" \
+        --prebuilt-core="${ocgcore}/bin" \
+        --sound=sfml
+    '';
+
+    enableParallelBuilding = true;
+    CXXFLAGS = "-include cstdint";
+    LDFLAGS = "-I ${irrlicht-edopro}/include -L ${irrlicht-edopro}/bin";
+    buildFlags = "verbose=true config=release_x64 ygoprodll";
+    makeFlags = "-C build";
+
+    # TODO find out how to get arch name from nix for cross compiled
+    installPhase = ''
+      mkdir -p $out/bin
+      cp bin/x64/release/ygoprodll $out/bin
+    '';
+  };
+
+  edopro-script =
+    let
+      assetsToCopy = lib.concatStringsSep "," [
+        # Needed if we download files from ProjectIgnis' sercet website
+        # "cacert.pem"
+        "config"
+        "deck"
+        "COPYING.txt"
+        "expansions"
+        "lflists"
+        "notices"
+        "puzzles"
+        "fonts"
+        "script"
+        "skin"
+        "sound"
+        "textures"
+        "WindBot"
+      ];
+    in
+    writeShellScriptBin "EDOPro" ''
+      set -eu
+      EDOPRO_DIR="$HOME/.local/share/edopro"
+
+      if [ ! -d $EDOPRO_DIR ]; then
+          mkdir -p $EDOPRO_DIR
+          cp -r ${assets.src}/{${assetsToCopy}} $EDOPRO_DIR
+
+          find $EDOPRO_DIR -type d -exec chmod u=rwx,go-rwx {} +
+          find $EDOPRO_DIR -type f -exec chmod u=rw,go-rwx {} +
+
+          rm $EDOPRO_DIR/config/io.github.edo9300.EDOPro.desktop.in
+      fi
+
+      export LD_LIBRARY_PATH=$EDOPRO_DIR:'${
+        lib.makeLibraryPath [
+          libGL
+          libX11
+          libxkbcommon
+          libXxf86vm
+          wayland
+          egl-wayland
+        ]
+      }';
+
+      exec ${edopro}/bin/ygoprodll -C $EDOPRO_DIR $@
+    '';
+
+  edopro-desktop = runCommandLocal "io.github.edo9300.EDOPro.desktop" { } ''
+    cp ${assets.src}/config/io.github.edo9300.EDOPro.desktop.in desktop-template
+
+    sed '/Path=/d' -i desktop-template
+    sed 's/Exec=.*/Exec=EDOPro/' -i desktop-template
+    sed 's/Icon=.*/Icon=EDOPro/' -i desktop-template
+
+    install -D desktop-template $out/share/applications/io.github.edo9300.EDOPro.desktop
+  '';
+in
+symlinkJoin {
+  name = "edopro-application";
+  paths = [
+    edopro
+    edopro-script
+    edopro-desktop
+  ];
+
+  postBuild = ''
+    mkdir -p $out/share/icons/hicolor/256x256/apps/
+    ${imagemagick}/bin/magick \
+        ${assets.src}/textures/AppIcon.png \
+        -resize 256x256 \
+        $out/share/icons/hicolor/256x256/apps/EDOPro.png
+  '';
+
+  meta = with lib; {
+    description = "The bleeding-edge automatic duel simulator, a fork of the YGOPro client.";
+    homepage = "https://projectignis.github.io";
+    changelog = "https://github.com/edo9300/edopro/releases";
+    license = licenses.agpl3Only;
+    maintainers = with maintainers; [ redhawk ];
+    platforms = [ "x86_64-linux" ];
```

Comment:
> optional: cross platform discussion

Sounds cool, but I have no hardware I could test anything else on at least. If you have a mac or raspberry pi to test this on, that's your call, but I don't think it needs to block this PR.

Reply:
Are there some nixos runners we could test on or something?

Reply:
Technically, but we shouldn't waste CI time, and it'll be pretty much impossible to debug on there if stuff stops working.

Reply:
That's fair, this is why I compile the core to. But we'll fix it when someone with the hardware that requires it wants to.